plugins/chat/assets/javascripts/discourse/components/chat-live-pane.js

```diff
@@ -1056,7 +1055,7 @@ export default Component.extend({
           upload_ids: (uploads || []).mapBy("id"),
         },
       }).then(() => {
-        this.onSwitchChannel(c);
+        this.router.transitionTo("chat.channel", "-", c.id);
```

Comment:
This is one of the big advantages of the work of the last months, everything is routable, we don't need all these actions everywhere, an URL is all you need!